topology/nhlt/nhlt-processor.c

```diff
@@ -265,6 +267,14 @@ int nhlt_create(struct tplg_pre_processor *tplg_pp)
 		eps_count++;
 	}
 
+	/* we can have 0 to several ssp eps */
+	for (i = 0; i < nhlt_ssp_get_ep_count(tplg_pp); i++) {
+		ret = nhlt_ssp_get_ep(tplg_pp, &eps[eps_count], i);
+		if (ret < 0)
+			goto err;
+		eps_count++;
+	}
+
```

Comment:
you should really have a comment in the earlier code that tells eps_count is modified in future patches... Or move the eps_count to when it's actually used, otherwise static analysis will have a field day.

Reply:
hmm yes, I'll run the patches through some tool and try to fix this